### ISSUE TITLE
False positive for MD053 and MD034

### ISSUE THREAD

**Issue by bronsen** [swanysimon/mdlint#53]
## Problem Description

The following markdown file is rejected by `mdlint check simple.md` but accepted by `markdownlint-cli2 simple.md`.

```markdown
# required title

Here a [reference] is used.

And below it is defined:

[reference]: https://example.com/
```
## mdlint Rejects

`mdlint --version` yields "mdlint 0.3.18"

Here is mdlint rejecting the file:

```text
bronsen@ananke in ~/C/l/teilchensammler-cli on  main via 🐍 (teilchensammler-cli) 
11:37:21 💚❯ mdlint check simple.md
simple.md
  simple.md:7:1: MD053 Link reference definition 'reference' is defined but not used
       | [reference]: https://example.com/
       | ^
  simple.md:7:14: MD034 Bare URL used: https://example.com/
       | [reference]: https://example.com/
       |              ^

Found 2 error(s) in 1 file(s) (1 checked)

```

## markdownlint-cli2 Accepts

`markdownlint-cli2 --help` yields "markdownlint-cli2 v0.23.0 (markdownlint v0.41.0)".

Here is markdownlint-cli2 accepting the file in question:

```text
bronsen@ananke in ~/C/l/teilchensammler-cli on  main via 🐍 (teilchensammler-cli) 
11:37:19 💚❯ markdownlint-cli2 simple.md
markdownlint-cli2 v0.23.0 (markdownlint v0.41.0)
Finding: simple.md
Linting: 1 file(s)
Summary: 0 error(s)

```

## Other Stuff

I believe the file simple.md to be correctly formatted.

**Comment by swanysimon** [comment]
Hey, thanks for the issue! None of the codebases I've used have had references yet, and you've found a bug in my parser. I see where I'll be replacing some regex (oops) I used for link parsing with some more properly parsed stuff from pulldown-cmark, but it might take a second to figure out everywhere that needs fixing. So right now: yes, bug.

**Comment by swanysimon** [comment]
Shipped in 0.3.21, please reopen if it isn't sufficient

**Comment by bronsen** [comment]
It has much improved, thank you for working on it :)

However, in a more complex markdown file I still get a false positive for MD053.

For this [README.md](https://code.c-base.org/infuanfu/teilchensammler-cli/raw/commit/7e7e115e592f3d106f2489902c97cacc80616637/README.md) mdlint 0.3.21 reports this:

    README.md:119:1: MD053 Link reference definition 'Textual' is defined but not used
       | [Textual]: https://textual.textualize.io/
       | ^

In fact, `[Textual]` is used twice, on lines 3 and 88.

----

I do not have permission to reopen the issue.

**Comment by swanysimon** [comment]
First time running a real project, apologies. I'll fix the permissions here one day :( 

I'll try to get this fixed tonight, I see where I need to make the change

**Comment by swanysimon** [comment]
<img width="762" height="884" alt="Image" src="https://github.com/user-attachments/assets/eaa757a6-4e34-46c1-b5c5-b28f77e89f5c" />

On the issue permission thing, I'll have to ask an LLM but the thing my human eyes found tells me it should have worked :(

**Comment by swanysimon** [comment]
@bronsen Could you try 0.3.22?

**Comment by bronsen** [comment]
Version 0.3.22 accepts myproject's README with regards to references.
